internal/credential/credential_libraries.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// LibraryRepository defines the interface expected
+// to get the total number of credential libraries and deleted ids.
+type LibraryRepository interface {
+	GetTotalLibraries(context.Context) (int, error)
+	GetTotalSSHCertificateLibraries(context.Context) (int, error)
+	ListDeletedLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListDeletedSSHCertificateLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+}
+
+// NewLibraryService returns a new credential library service.
```

Comment:
This is also a departure from what has been done previously in that this is creating a struct for the service, while previously services where implemented as a function. I think it is fine to have a struct, at least if we really need multiple methods, but we might want to pick a name that is more descriptive of the particular service that is being implemented, rather than claiming the "LibraryService" name.

Reply:
This was originally named `CredentialLibraryService` and Mike suggested dropping the stuttering since it was `credential.CredentialLibraryService`. Now it's `credential.LibraryService`. Do you have another suggestion?

Reply:
I've renamed the files these services are defined in, thanks